### PR TITLE
Maint/pre commit

### PR DESCRIPTION
**Pull request by justinGilmer**
### PR Summary:
Foyer currently lacks an automated way to enforce a style guide and
basic docstring formatting that is easily distributed to other
developers.

An example of this is mBuild, which now uses the `pre-commit` framework
to automate this during pull-requests as well as an install-able git
commit hook.
This commit adds the initial configuration files for pre-commit setup.

Also, the build and recipes for the `devtools/conda-recipe/` folder. Since foyer is now hosted on conda-forge, this is not needed.

### PR Checklist
------------
 - [x] Includes appropriate unit test(s)
 - [x] Appropriate docstring(s) are added/updated
 - [x] Code is (approximately) PEP8 compliant
 - [ ] Issue(s) raised/addressed?

Still need to enable the `pre-commit pr` extension like we did for mBuild.

### CODE REVIEW

**Review by jennyfothergill — COMMENTED** [review]
Nice work! :) I reviewed with "hide whitespace" option with particular focus on new docstrings (as I am assuming most changes within the code are applied automatically) and found no typos. 
Requested changes:
- [ ] enable [pre-commit.ci](https://results.pre-commit.ci/) to ensure that future PRs keep consistent style.
- [ ] add a brief explanation somewhere in docs and/or contributing guide of how to use pre-commit.

File: foyer/smarts_graph.py [hunk 1]
```diff
@@ -23,7 +24,7 @@ class SMARTSGraph(nx.Graph):
         Rules or SMARTSGraph over which this SMARTSGraph takes precedence
 
     Other Parameters
-    ----------
+    ----------------
     args
     kwargs
```
[the remark this answers is not in the document]
Comment (justinGilmer) [reply foyer/smarts_graph.py:29]: Good catch! Added a link and small `Notes` section in the docstring.

File: foyer/utils/io.py [hunk 2]
```diff
@@ -1,47 +1,51 @@
...
         The module you'd like to import, as a string
+
     Returns
     -------
     module : {module, object}
         The module object
+
     Examples
     --------
     >>> # the following two lines are equivalent. the difference is that the
```
[the remark this answers is not in the document]
Comment (justinGilmer) [reply foyer/utils/io.py:51]: yeah this is a method that is normally used under-the-hood, so i think its fine to have it here, and not worry about rendering it.

**Review by daico007 — APPROVED** [review]
LGTM!

**Review by jennyfothergill — APPROVED** [review]
LGTM!

**Pull request merged**